docs/userguide/storagedriver/device-mapper-driver.md

```diff
@@ -397,6 +397,176 @@ there are two key directories. The `/var/lib/docker/devicemapper/mnt` directory
+
+```bash
+$ sudo docker info
+Containers: 0
+ Running: 0
+ Paused: 0
+ Stopped: 0
+Images: 2
+Server Version: 1.11.0-rc2
+Storage Driver: devicemapper
+ Pool Name: docker-8:1-123141-pool
+ Pool Blocksize: 65.54 kB
+ Base Device Size: 10.74 GB
+ Backing Filesystem: ext4
+ Data file: /dev/loop0
+ Metadata file: /dev/loop1
+ Data Space Used: 1.202 GB
+ Data Space Total: 107.4 GB
+ Data Space Available: 4.506 GB
+ Metadata Space Used: 1.729 MB
+ Metadata Space Total: 2.147 GB
+ Metadata Space Available: 2.146 GB
+ Udev Sync Supported: true
+ Deferred Removal Enabled: false
+ Deferred Deletion Enabled: false
+ Deferred Deleted Device Count: 0
+ Data loop file: /var/lib/docker/devicemapper/devicemapper/data
+ WARNING: Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.
+ Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
+ Library Version: 1.02.90 (2014-09-01)
+Logging Driver: json-file
+[...]
+```
+
+The `Data Space` values show that the pool is 100GB total. This example extends the pool to 200GB.
+
+1. List the sizes of the devices.
+
+	```bash
+	$ sudo ls -al /var/lib/docker/devicemapper/devicemapper/
+	total 1175492
+	drwx------ 2 root root         4096 Mar 29 02:45 .
+	drwx------ 5 root root         4096 Mar 29 02:48 ..
+	-rw------- 1 root root 107374182400 Mar 30 05:22 data
+	-rw------- 1 root root   2147483648 Mar 31 11:17 metadata
+	```
+
+2. Truncate `data` file to the size of the `metadata` file (approximate 200GB).
+
+	```bash
+	$ sudo truncate -s 214748364800 /var/lib/docker/devicemapper/devicemapper/data
+	```
+
+3. Verify the file size changed.
+
+	```bash
+	$ sudo ls -al /var/lib/docker/devicemapper/devicemapper/
+	total 1175492
+	drwx------ 2 root root         4096 Mar 29 02:45 .
+	drwx------ 5 root root         4096 Mar 29 02:48 ..
+	-rw------- 1 root root 214748364800 Mar 31 11:20 data
+	-rw------- 1 root root   2147483648 Mar 31 11:17 metadata
+	```
+
+4. Reload data loop device
+
+	```bash
+	$ sudo blockdev --getsize64 /dev/loop0
+	107374182400
+	$ sudo losetup -c /dev/loop0
+	$ sudo blockdev --getsize64 /dev/loop0
+	214748364800
+	```
+
+5. Reload devicemapper thin pool.
+
+	a. Get the pool name first.
+
+		$ sudo dmsetup status | grep pool
+		docker-8:1-123141-pool: 0 209715200 thin-pool 91 422/524288 18338/1638400 - rw discard_passdown queue_if_no_space -
+
+		The name is the string before the colon.
+
+	b. Dump the device mapper table first.
+
+		$ sudo dmsetup table docker-8:1-123141-pool
+		0 209715200 thin-pool 7:1 7:0 128 32768 1 skip_block_zeroing
+
+	c. Calculate the real total sectors of the thin pool now.
+
+		Change the second number of the table info (i.e. the disk end sector) to reflect the new number of 512 byte sectors in the disk. For example, as the new loop size is 200GB, change the second number to 419430400.
+
```

Comment:
I think second number reflects number of sectors and not the end sector. Look at "man dmsetup" in section TABLE FORMAT.